src/neo/SmartContract/Native/GasToken.cs

```diff
@@ -22,8 +23,10 @@ internal override void Initialize(ApplicationEngine engine)
         internal override void OnPersist(ApplicationEngine engine)
         {
             long totalNetworkFee = 0;
+            HashSet<UInt160> distributed = new HashSet<UInt160>();
             foreach (Transaction tx in engine.PersistingBlock.Transactions)
             {
+                if (distributed.Add(tx.Sender)) NEO.DistributeGas(engine, tx.Sender);
```

Comment:
This is just to ensure that it is distributed only once per `Persist`, right?

Reply:
Isn't there any need to avoid the https://github.com/neo-project/neo/blob/02fdcc62d5a13f5ff2e39a8468ac4cd6f463b884/src/neo/SmartContract/Native/NeoToken.cs#L45?

If the transfer also changes NEO's balance?

Reply:
We're running this line in `OnPersist` context. It invokes `GAS.Mint`. It calls `PostTransfer`. Which calls `onNEP17Payment`. That fails. What are we gonna do then? Fail persisting a block? Not an option. Ignore this problem and fail GAS distribution? Fine, except some storage changes are already in-flight in this context and, most importantly, `tx.Sender` may just not have enough GAS for the `Burn()` one line below. So, what are the options we have?

Reply:
@roman-khimov it's the same problem here https://github.com/neo-project/neo/blob/bfbb78b0e7b6050e03376d4e72e50ecae841c1c3/src/neo/SmartContract/Native/NeoToken.cs#L161 , I think that we should not rise `onNEP17Payment` during `OnPersist`

Reply:
It's not. This invocation specifically has `false` for `callOnPayment`. And we only have simple standard accounts in the committee (their keys are known).

Reply:
I'll put "false" here too

Reply:
I don't think it's legal to do this! Unlike committee members transaction sender can be anything, including deployed contract. And deployed contract must be invoked on this mint.

Reply:
Yes, I we merge it, the generated gas will come to not call `onNEP17Payment`, before this PR it does....